tests/t500-operator.cu

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
```

Comment:
This looks so close to the CPU code for the QFunction and close to the OCCA version. I really want to start prototyping some approaches to getting down to one source file for the QFunctions - I know users will make bugs juggling 3 copies of the QFunction because I make bugs juggling 2 copies.

I am ok merging the use of a .cu file for the QFunctions as well only under the understanding that we will work to remove this soon and get down to 1 source file for QFunctions.

Reply:
The case where it will be different, is for non-linear operators where we compute the geometric factors on the fly. Then using GPU specific code will be unavoidable I think, but the GPU code can easily degenerate to CPU code I think. I'm just a bit worried about memory coalescing, this might require code that is only specific to GPU to load data efficiently, as incorrect memory access easily result in a factor 30 slowdown.

Reply:
Geometry on the fly vs precomputed will require completely different QFunctions and Operators. I think it makes the case for single source files for QFunctions even stronger as it would be better for performance comparisons to swap out different single implimentations of particular operators rather than having to swap out triplicates of the operators and not knowing 100% that they are identical in their action.